navigation/infomax.h

```diff
@@ -203,27 +203,56 @@ class InfoMaxRotater : public InfoMax<FloatType>
     //------------------------------------------------------------------------
     // Public API
     //------------------------------------------------------------------------
+    template<class... Ts>
+    const std::vector<FloatType> &getImageDifferences(Ts &&... args) const
+    {
+        calcImageDifferences(std::forward<Ts>(args)...);
+        return m_RotatedDifferences;
+    }
+
     template<class... Ts>
     auto getHeading(Ts &&... args) const
```

Comment:
Does this need a ``&`` to make sure we get a reference and not a copy?

Reply:
Where exactly? possible places to put ``&`` is almost as fun a C++ quiz question as possible places to put ``const``...

Reply:
Oh yeah, sorry :-p

I *think* the ``auto`` in the return type might be resolved to a value type cf. a reference.

I meant like this:
``auto &getHeading(Ts &&... args) const``

It also looks like you can use ``decltype(auto)`` for this:
https://en.cppreference.com/w/cpp/language/auto

Reply:
Ahh ok - I don't **think** ``getHeading`` should return a reference though - it's returning a tuple which, amongst other things, include a reference to ``m_RotatedDifferences``.

Reply:
Ah, my bad. I guess in that case you def won't want to return by reference.